os/log.c

```diff
@@ -227,6 +227,63 @@ static void initSyslog(void) {
 #endif
 }
 
+static void
+LogFailedWriteStdout(const void * buf, size_t len)
+{
+        if (write(STDOUT_FILENO, buf, len)==-1)
+        {
+	    /* We can't write to the logfile, stderr, and stdout; something
+	     * really bad is happening. */
+            AbortServer();
```

Comment:
I'm not sure whether that's a good idea. In certain embedded scenarios, those things might well happen, even on purpose. 

IMHO, it should be enough enough trying to write to stderr when logfile can't be written (eg. disk full or no logfile at all). But if stderr isn't available, then stdout most likely isn't too. And that in turn can only happen on purpose. Don't see any hard reason to kill the Xserver here.